crates/motionfile/src/motion_interpolator.rs

```diff
@@ -0,0 +1,157 @@
+use std::fmt::Debug;
+use std::time::Duration;
+
+use crate::spline_interpolator::SplineInterpolator;
+use crate::MotionFileFrame;
+use crate::{condition::ConditionEnum, Condition, MotionFile};
+use color_eyre::eyre::Context;
+use color_eyre::{Report, Result};
+use splines::{Interpolate, Interpolation, Key};
+use types::SensorData;
+
+pub struct MotionInterpolator<T>
+where
+    T: Debug + Interpolate<f32>,
```

Comment:
Not needed here, as it's already defined in L38

Reply:
Done